src/sdk/makeChart.js

```diff
@@ -258,5 +270,49 @@ export default ({ sdk, parent, getChart = fetchChartData, chartsMetadata, attrib
     getClosestRow,
   }
 
-  return { ...instance, ...makeDimensions(instance), ...makeFilterControllers(instance), destroy }
+  onKeyChange(["Alt", "Shift", "KeyF"], () => {
+    node.updateAttribute("fullscreen", !node.getAttribute("fullscreen"))
+  })
+
+  const dimensions = makeDimensions(instance)
+
+  const onDimensionToggle = onKeyAndMouse(
+    ["Shift"],
+    id => ({ allPressed: merge }) => dimensions.toggleDimensionId(id, { merge }),
+    { allPressed: false }
+  )
+
+  const onHighlight = onKeyAndMouse("Alt", () => ({ allPressed }) =>
+    allPressed &&
+    node.updateAttributes({
+      navigation: "highlight",
+      prevNavigation: node.getAttribute("navigation"),
+    })
+  )
+
+  const onSelectVerticalAndZoom = onKeyAndMouse(["Shift", "Alt"], () => ({ allPressed }) =>
+    allPressed &&
+    node.updateAttributes({
+      navigation: "selectVertical",
+      prevNavigation: node.getAttribute("navigation"),
+    })
+  )
+
+  const onSelectAndZoom = onKeyAndMouse("Shift", () => ({ allPressed }) =>
+    allPressed &&
+    node.updateAttributes({ navigation: "select", prevNavigation: node.getAttribute("navigation") })
+  )
+
+  return {
+    ...instance,
+    ...dimensions,
+    ...makeFilterControllers(instance),
+    destroy,
+    onKeyChange,
+    onKeyAndMouse,
+    onDimensionToggle,
+    onHighlight,
+    onSelectAndZoom,
+    onSelectVerticalAndZoom,
```

Comment:
This is a personal point of view. The `makeChart` is trying to be independent from the UI, the UI may be a pie chart or a gauge that does not contain shortcuts. I would prefer to move the keyboard logic in a plugin-like implementation. So the makeChart does not depend on the shortcuts. Something like

```
// src/chartLibraries/dygraph/navigation/generic.js
chartUI.trigger("mousedown")
```

```
// src/sdk/plugins/keyboard
export default sdk => {
let off

  sdk.on("focus", chart => {
    if (chart is not dygraph) return
    addKeyboardListeners()

   off = chart.getUI().on("mousedown", ...).on('wheel', (g, percentage, xPct) => zoom(g, percentage, xPct)).on(...)
  })
  
  sdk.on("blur", chart => {
    // clearKeyboard, etc.
    if (off) off()
  })
}
```